open-company/open-company-auth
# Slack invite

This PR for adds to the Auth service:

- Uninvited Slack users to the roster listing
- Invite uninvited Slack user via Slack (SQS request to the bot)
- OC Team created by a Slack user inherits the Slack team's logo as team `logo-url` property

Make sure you pull infrastructure repo. It has a mainline change needed for Auth.

This PR works in conjunction with web changes and [this bot PR](https://github.com/open-company/open-company-bot/pull/15).

To test:

- Use updated bot and web branches
- Onboard a brand new OC user/org w/ Slack, ensure the team listing and team responses have the Slack team's logo as `logo-url`
- Pull the roster of the OC team, ensure you see all the members of the Slack team except the creator in the response with a status of `uninvited`
- Invite one of the Slack users using the invite request, check the following:
   - do you get a new user in the Auth DB w/ status `pending`?
   - do you get an SQS invite message in the bot queue?
   - when you pull the roster, is the invited user listed as `pending` and not there as `uninvited`?
- Merge
- Deploy w/ [bot](https://github.com/open-company/open-company-bot/pull/15), web
- Drop and give me 20 pushups
